### ISSUE TITLE
keysOf

### ISSUE THREAD

**Issue by Undistraction** [char0n/ramda-adjunct#366]
keysOf :: [String] => Object => [String]

Ramda currently offers [props](http://ramdajs.com/docs/#props), but there is no equivalent for keys, so how about `keysOf`:

```
const keysOf = curry((keys, o) => compose(filter(flip(has)(o)))(keys))

const o = {
  a: 1,
  b: 2,
  c: 3,
}

const oKeys = [
  'a', 'c', 'd', 'e', 'f'
]

keysOf(oKeys, o)
```
[REPL](http://ramdajs.com/repl/?v=0.25.0#?const%20keysOf%20%3D%20curry%28%28keys%2C%20o%29%20%3D%3E%20compose%28filter%28flip%28has%29%28o%29%29%29%28keys%29%29%0A%0Aconst%20o%20%3D%20%7B%0A%20%20a%3A%201%2C%0A%20%20b%3A%202%2C%0A%20%20c%3A%203%2C%0A%7D%0A%0Aconst%20oKeys%20%3D%20%5B%0A%20%20%27a%27%2C%20%27c%27%2C%20%27d%27%2C%20%27e%27%2C%20%27f%27%0A%5D%0A%0AkeysOf%28oKeys%2C%20o%29)

Note: Ramda's props returns `undefined` values for any values that aren't matched, but I don't think that makes sense here as these are keys, not the values. 

Usecase: I need to make sure an object doesn't include more than one of a set of exclusive keys.

**Comment by char0n** [comment]
http://ramdajs.com/docs/#pick, http://ramdajs.com/docs/#pickAll or http://ramdajs.com/docs/#pickBy doesn't satisfy your usecase ?

**Comment by Undistraction** [comment]
> http://ramdajs.com/docs/#pick, http://ramdajs.com/docs/#pickAll or http://ramdajs.com/docs/#pickBy doesn't satisfy your usecase ?

No. They all have a very different intent - they are focused on copying key value pairs from an object to a new object - `Object -> ... -> Object`. The function I'm talking about uses an object to filter a list of keys  `[String] -> ... -> [String]`

**Comment by char0n** [comment]
Ahh I get it. You want intersection of the list of the possible keys and the real ones.

**Comment by char0n** [comment]
intersectKeysWith = curry(fn, keys, obj) => ...);
IntersectKeys = intersectKeysWith(has);

Does it make sence ?

**Comment by Undistraction** [comment]
Makes sense: [REPL](http://ramdajs.com/repl/?v=0.25.0#?const%20intersectKeysWith%20%3D%20curry%28%28fn%2C%20keys%2C%20o%29%20%3D%3E%20compose%28filter%28flip%28fn%29%28o%29%29%29%28keys%29%29%0A%0Aconst%20intersectKeys%20%3D%20intersectKeysWith%28has%29%0A%0Aconst%20o%20%3D%20%7B%0A%20%20a%3A%201%2C%0A%20%20b%3A%202%2C%0A%20%20c%3A%203%2C%0A%7D%0A%0Aconst%20oKeys%20%3D%20%5B%0A%20%20%27a%27%2C%20%27c%27%2C%20%27d%27%2C%20%27e%27%2C%20%27f%27%0A%5D%0A%0AintersectKeys%28oKeys%2C%20o%29%20%2F%2F%20%5B%22a%22%2C%20%22c%22%5D)

**Comment by GingerPlusPlus** [comment]
Sounds to me like `R.difference(oKeys, R.keys(o)`, or, as a function, `R.useWith(R.difference, [R.identity, R.keys])`
